code/datums/martial/buster_style.dm

```diff
@@ -228,74 +213,84 @@
 				tossedliving.visible_message(span_warning("[user] throws [tossedliving] by [tossedliving.p_their()] tail, severing [tossedliving.p_them()] from it!")) //"I'm taking this back."
 				T.Remove(tossedliving)
 				user.put_in_hands(T)
-	user.visible_message(span_warning("[user] throws [tossed]!"))
-	for(var/i = 1 to target_dist)
-		var/dir_to_target = get_dir(get_turf(tossed), D) //vars that let the thing be thrown while moving similar to things thrown normally
-		var/turf/T = get_step(get_turf(tossed), dir_to_target)
-		if(T.density) // crash into a wall and damage everything flying towards it before stopping 
+	user.visible_message(span_warning("[user] throws [thrown]!"))
+	flying |= thrown
+	soar(user, thrown, D, target_dist, flying)
+
+
+
+
+
+
+/datum/martial_art/buster_style/proc/soar(mob/living/user, atom/movable/center, turf/endzone, distance = 0, list/thrown) //proc for throwing something you picked up with grapple
+	var/slamdam = 7
+	var/objdam = 50
+	if(distance == 0)
+		return
+	var/dir_to_target = get_dir(get_turf(center), endzone) //vars that let the thing be thrown while moving similar to things thrown normally
+	var/turf/next = get_step(get_turf(center), dir_to_target)
+	if(next.density) // crash into a wall and damage everything flying towards it before stopping 
+		for(var/mob/living/S in thrown)
```

Comment:
You have two variables named thrown
A list thats specific to the proc
And a variable thats for the entire martial art